cdap-ui/app/cdap/components/ResourceCenterEntity/ResourceCenterPipelineEntity.js

```diff
@@ -0,0 +1,133 @@
+  const iconClassName = 'icon-pipelines';
+
+  const resourceCenterId = uuid.v4();
+  const hydratorLinkStateName = 'hydrator.create';
+  const hydratorLinkStateParams = {
+    namespace: NamespaceStore.getState().selectedNamespace,
+    artifactType: 'cdap-data-pipeline'
+  };
+
+  const hydratorCreateLink = window.getHydratorUrl({
+    stateName: hydratorLinkStateName,
+    stateParams: hydratorLinkStateParams
+  });
+  const hydratorImportLink = window.getHydratorUrl({
+    stateName: hydratorLinkStateName,
+    stateParams: {
+      ...hydratorLinkStateParams,
+    resourceCenterId
+    }
+  });
+
+  const createBtnHandler = () => {
+    PlusButtonStore.dispatch({
+      type: 'TOGGLE_PLUSBUTTON_MODAL',
+      payload: {
+        modalState: false
+      }
+    });
+  };
+
+  const importBtnHandler = (event) => {
+    if (!objectQuery(event, 'target', 'files', 0)) {
+      return;
+    }
+
+    let uploadedFile = event.target.files[0];
+
+    if (uploadedFile.type !== 'application/json') {
+      // TODO: Currently no way to show this in Resource Center modal yet
+      console.error('Pipeline configuration should be JSON.');
+      return;
+    }
+
+    // still have to do this because JSON.stringify(uploadedFile) doesn't work
+    let reader = new FileReader();
+    reader.readAsText(uploadedFile, 'UTF-8');
+
+    reader.onload =  (evt) => {
+      let fileDataString = evt.target.result;
+      window.localStorage.setItem(resourceCenterId, fileDataString);
+      window.location.href = hydratorImportLink;
+    };
+  };
+
+  return (
+    <div className={classnames('resourcecenter-entity-card resourcecenter-entity-card-pipeline', className)}>
+      <div className="image-content-container">
+        <div className="image-container">
+          <div className="entity-image">
+            <IconSVG name={iconClassName} />
+          </div>
+        </div>
+        <div className="content-container">
+          <div className="content-text">
+            <h4>{title}</h4>
+            <p>{description}</p>
+          </div>
+        </div>
+      </div>
+      <div className="buttons-container">
+        <a href={hydratorCreateLink}>
+          <button
+            id={(primaryActionLabel + "-" + title).toLowerCase()}
+            className="btn btn-primary"
+            onClick={createBtnHandler}
+          >
+            {primaryActionLabel}
+          </button>
+        </a>
+        <Label
+          for="resource-center-import-pipeline"
+          id={(secondaryActionLabel + "-" + title).toLowerCase()}
+          className="btn btn-secondary"
+        >
+          {secondaryActionLabel}
+          <Input
+            type="file"
+            accept='.json'
+            id="resource-center-import-pipeline"
+            onChange={importBtnHandler}
+          />
+        </Label>
+      </div>
+    </div>
+  );
+}
+ResourceCenterPipelineEntity.propTypes = {
+  className: PropTypes.string
+};
```

Comment:
it doesn't seem like you are passing any class name to the props of this element. Is this still needed?